hyper-boring/src/lib.rs

```diff
@@ -230,8 +230,17 @@ where
 
             let host = uri.host().ok_or("URI missing host")?;
 
-            let config = inner.setup_ssl(&uri, host)?;
-            let stream = tokio_boring::connect(config, host, conn).await?;
+            // If `host` is an IPv6 address, we must strip away the square brackets that surround
+            // it (otherwise, boring will fail to parse the host as an IP address, eventually
+            // causing the handshake to fail due a hostname verification error).
+            let host_or_ip = match (host.find('['), host.find(']')) {
+                (Some(0), Some(i)) => &host[1..i],
```

Comment:
We need to ensure here that `i == host.len() - 1`. Otherwise, we can strip a part of host, doesn't seem to be a safe thing to do. Should be smthg like:
```rust
let last = host.len() - 1;

if last > 0 && host[0] == '[' && host[last] == ']' {
    host = host [1..last];
}
```
Ideally we should also validate that what's enclosed in `[ ]` is indeed and IPv6

Reply:
Good point (`host` is extracted from a valid `Uri`, so I assumed it could only contain brackets if it is a valid IPv6 address). That being said I'm happy to make this change.